watertap/unit_models/tests/test_surrogate_crystallizer.py

```diff
@@ -0,0 +1,229 @@
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+import os
+import pytest
+from pyomo.environ import (
+    ConcreteModel,
+    Var,
+    Constraint,
+    units as pyunits,
+    assert_optimal_termination,
+)
+from pyomo.util.check_units import assert_units_consistent
+from idaes.core import FlowsheetBlock
+from idaes.core.util.exceptions import ConfigurationError
+from watertap.property_models.water_prop_pack import WaterParameterBlock
+from watertap.property_models.multicomp_aq_sol_prop_pack import (
+    MCASParameterBlock,
+    MaterialFlowBasis,
+)
+from idaes.core.util.model_statistics import degrees_of_freedom
+
+from idaes.core.surrogate.surrogate_block import SurrogateBlock
+from idaes.core.surrogate.pysmo_surrogate import (
+    PysmoSurrogate,
+)
+from watertap.unit_models.surrogate_crystallizer import SurrogateCrystallizer
+
+from idaes.core import UnitModelCostingBlock
+from watertap.costing import WaterTAPCosting
+from watertap.core.solvers import get_solver
+
+__author__ = "Oluwamayowa Amusat, Adam Atia"
+
+
+# TODO: can consider adding NN model surrogate function and test
+def add_crystallizer_rbf_model(
+    blk,
+    surrogate_inputs_with_bounds,
+    surrogate_outputs,
+    surrogate_to_flowsheet_basis_ratio,
+):
+    ##############################################################################################################
+    # block loading into IDAES Surrogate Block
+    ##############################################################################################################
+    crystallizer_inputs = [
+        surrogate_inputs_with_bounds[k]["flowsheet_var"]
+        for k in surrogate_inputs_with_bounds.keys()
+    ]
+    crystallizer_outputs = [
+        surrogate_outputs[k]["flowsheet_var"] for k in surrogate_outputs.keys()
+    ]
+
+    try:
+        assert set(list(blk.solids_list)).issubset(surrogate_outputs.keys())
+    except AssertionError:
+        raise ConfigurationError(
+            "List of solids provided in crystallizer model must match surrogate output keys for solids. Please check."
+        )
+
+    filename = [
+        "Vapor_Pressure",
+        "Calcite_g",
+        "Anhydrite_g",
+        "Glauberite_g",
+        "Halite_g",
+    ]
+    surrogate_directory = os.path.dirname(os.path.abspath(__file__))
+    surrogate_directory = os.path.join(
+        surrogate_directory, "surrogate_crystallizer_defaults"
+    )
+
+    for sm in range(0, len(filename)):
+        block_name = "crystallizer_surrogate" + "_" + filename[sm]
+        blk.add_component(block_name, SurrogateBlock(concrete=True))
+
+        surrogate_name = f"{filename[sm]}.json"
+        current_surrogate_filename = os.path.join(surrogate_directory, surrogate_name)
+        current_surrogate = PysmoSurrogate.load_from_file(
+            os.path.normpath(current_surrogate_filename)
+        )
```

Comment:
```suggestion
        current_surrogate = PysmoSurrogate.load_from_file(current_surrogate_filename)
```

Reply:
This was my previous attempt to deal with the path issues between platforms. Replacing this now